plugin/metrics/metrics.go

```diff
@@ -22,6 +23,7 @@ type Metrics struct {
 	ln      net.Listener
 	lnSetup bool
 	mux     *http.ServeMux
+	srv     *http.Server
 
```

Comment:
Adding the srv, allow to call the Shutdown() and ensure the service is really closed when shutting down the plugin.
now expected by UT.